src/deploy/functions/deploy.ts

```diff
@@ -22,16 +23,20 @@ async function uploadSourceV1(context: args.Context): Promise<void> {
     file: context.functionsSource!,
     stream: fs.createReadStream(context.functionsSource!),
   };
-  await gcs.upload(uploadOpts, uploadUrl);
+  await gcs.upload(uploadOpts, uploadUrl, {
+    "x-goog-content-length-range": "0,104857600",
+  });
 }
 
-async function uploadSourceV2(context: args.Context): Promise<void> {
-  const bucket = "staging." + (await gcs.getDefaultBucket(context.projectId));
+async function uploadSourceV2(context: args.Context, region: string): Promise<void> {
+  const res = await gcfv2.generateUploadUrl(context.projectId, region);
   const uploadOpts = {
     file: context.functionsSource!,
     stream: fs.createReadStream(context.functionsSource!),
   };
-  context.storageSource = await gcs.uploadObject(uploadOpts, bucket);
+  await gcs.upload(uploadOpts, res.uploadUrl);
+  context.storage = context.storage || {};
+  context.storage[region] = res.storageSource;
```

Comment:
nit* Not sure if I prefer this, but a single-liner:

```js
context.storage = { ...context.storage, ...{[region]: res.storageSource}};
```

Reply:
There's a simpler one-liner:

```
context.storage = {...context.storge, [region]: res.storageSource};
```

It definitely feels foreign to me, but I guess it's not bad. I can change.